UnityProject/Assets/Scripts/Systems/Antagonists/Objectives/Assassinate.cs

```diff
@@ -46,17 +46,17 @@ protected override void Setup()
 			}
 
 			// Pick a random target and add them to the targeted list
-			Target = playerPool.PickRandom().Script;
+			Target = playerPool.PickRandom().Script.connectedPlayer;
```

Comment:
Worth saying this will mean if someone respawns as a ghost role or is admin respawned then this will fail the objective.

Not much can be done about that without a big change

Reply:
Can't we just make a bool that is like `ObserverOrGhostRole`?
I don't think that's a big change, we just need to tell the connectedPlayer when we have rejoined as a ghost role after death.

Reply:
I think SS13 already does that as well because i don't think you can respawn as a normal character after you've chosen to play as a ghost role.

Reply:
hey, no need to even do a Boolean and keep updating/tracking it, I made a function that checks if the player is a ghost role.

Reply:
remember it has to be thread safe

Reply:
what?

Reply:
![image](https://user-images.githubusercontent.com/34368774/145443672-3ad42f49-766d-409b-ab5a-5fd6b57b9700.png)